components/tikv_util/src/interval_driver.rs

```diff
@@ -70,8 +69,13 @@ impl<K: KvEngine, R: RaftEngine> MetricsFlusher<K, R> {
         }
         drop(self.sender.take().unwrap());
         if let Err(e) = h.unwrap().join() {
-            error!("join metrics flusher failed"; "err" => ?e);
-            return;
+            error!("join thread failed"; "err" => ?e, "name" => self.name.as_str());
         }
     }
 }
+
+pub trait IntervalRunnable: Send + Sync {
```

Comment:
why not use `Runnable`?

Reply:
`Runnable` is a machine abstraction, `IntervalRunnable` is a task abstraction, I'll rename it to `IntervalTask` instead.